spyre-rag/src/retrieve/backend_server.py

```diff
@@ -145,7 +147,10 @@ def chat_completion():
             return jsonify({"error": "Server busy. Try again shortly."}), 429
 
         try:
+            start_time = time.time()
             vllm_stream = query_vllm_stream(prompt, docs, llm_endpoint, llm_model, stop_words, max_tokens, temperature, stream, dynamic_chunk_truncation=TRUNCATION)
```

Comment:
This call to `query_vllm_stream` in streaming mode returns a generator and not a final answer. This generator (like a C++ iterator) is then read chunk by chunk with each piece being sent to Web UI even after this Python backend function returns. So if you want to measure properly you should probably test non-streaming case only.

Reply:
@Niharika0306 chime in please to confirm.